lib/test-block-patterns.php

```diff
@@ -0,0 +1,204 @@
+		<!-- /wp:column -->
+		<!-- wp:column -->
+		<div class="wp-block-column"><!-- wp:site-tagline /--></div>
+		<!-- /wp:column -->
+		</div>
+		<!-- /wp:columns -->',
+	)
+);
+register_block_pattern(
+	'header/v2',
+	array(
+		'title'   => __( 'Header v2', 'gutenberg' ),
+		'scope'   => array(
+			'inserter'  => false,
+			'transform' => array( 'core/template-part/header' ),
+		),
+		'content' => '
+		<!-- wp:heading {"textAlign":"center","backgroundColor":"gray","textColor":"yellow"} -->
+		<h2 class="has-text-align-center has-yellow-color has-gray-background-color has-text-color has-background">This is the Header</h2>
+		<!-- /wp:heading -->
+		<!-- wp:columns {"align":"wide"} -->
+		<div class="wp-block-columns alignwide"><!-- wp:column -->
+		<div class="wp-block-column"><!-- wp:site-title /--></div>
+		<!-- /wp:column -->
+		<!-- wp:column -->
+		<div class="wp-block-column">
+		<!-- wp:site-tagline /-->
+		</div>
+		<!-- /wp:column --></div>
+		<!-- /wp:columns -->',
+	)
+);
+register_block_pattern(
+	'footer/v1',
+	array(
+		'title'   => __( 'Footer v1', 'gutenberg' ),
+		'scope'   => array(
+			'inserter'  => false,
+			'transform' => array( 'core/template-part/footer' ),
+		),
+		'content' => '
+		<!-- wp:heading {"textAlign":"center","backgroundColor":"gray","textColor":"yellow"} -->
+		<h2 class="has-text-align-center has-yellow-color has-gray-background-color has-text-color has-background">This is a Footer</h2>
+		<!-- /wp:heading -->
+		<!-- wp:columns {"align":"wide"} -->
+<div class="wp-block-columns alignwide"><!-- wp:column -->
+<div class="wp-block-column"><!-- wp:paragraph -->
+<p><a href="mailto:#">example@example.com<br></a>T. +00 (0)1 22 33 44 55</p>
+<!-- /wp:paragraph --></div>
+<!-- /wp:column -->
+<!-- wp:column -->
+<div class="wp-block-column"><!-- wp:paragraph {"align":"center"} -->
+<p class="has-text-align-center">2, Rue Losuis-Boilly<br>Paris, France</p>
+<!-- /wp:paragraph --></div>
+<!-- /wp:column -->
+<!-- wp:column {"verticalAlignment":"center"} -->
+<div class="wp-block-column is-vertically-aligned-center"><!-- wp:social-links {"align":"right","className":"is-style-twentytwentyone-social-icons-color"} -->
+<ul class="wp-block-social-links alignright is-style-twentytwentyone-social-icons-color"><!-- wp:social-link {"url":"https://wordpress.org","service":"wordpress"} /-->
+<!-- wp:social-link {"url":"https://www.facebook.com/WordPress/","service":"facebook"} /-->
+<!-- wp:social-link {"url":"https://twitter.com/WordPress","service":"twitter"} /-->
+<!-- wp:social-link {"url":"https://www.youtube.com/wordpress","service":"youtube"} /--></ul>
+<!-- /wp:social-links --></div>
+<!-- /wp:column --></div>
+<!-- /wp:columns -->',
+	)
+);
+
+
+// Tests with InnerBlocks like Buttons.
+register_block_pattern(
+	'buttons/rigas',
+	array(
+		'title'   => __( 'Buttons v1', 'gutenberg' ),
+		'scope'   => array(
+			'inserter'  => false,
+			'transform' => array( 'core/buttons' ),
```

Comment:
Is this like a shortcut to avoid parsing the pattern? Should it be just `blockTypes` at the top level of the pattern?

Reply:
I've created this PR to remove `scope`: https://github.com/WordPress/gutenberg/pull/30471